### PR TITLE
changes done in framework while developing SAML Inbound Authenticator

### PR DESCRIPTION
**Pull request by ChamaraPhilipsuom**

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/inbound/IdentityServlet.java [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 package org.wso2.carbon.identity.application.authentication.framework.inbound;
 
-import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang.StringUtils;
```
Comment (omindu) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/inbound/IdentityServlet.java]: Any reason for changing this to lang3 from lang2 ?

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java [hunk 2]
```diff
@@ -781,25 +781,22 @@ public static String getHostName() {
...
+                } else {
+                    queryString.append("&");
                 }
-                queryString.append("&");
-                queryString.append(entry.getKey());
+                queryString.append(URLEncoder.encode(entry.getKey(), "UTF-8"));
                 queryString.append("=");
-                queryString.append(paramValue);
-
+                queryString.append(URLEncoder.encode(paramValue, "UTF-8"));
```
Comment (omindu) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java]: We can use `StandardCharsets.UTF_8`.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java [hunk 3]
```diff
@@ -361,9 +361,9 @@ public int compare(IdentityProcessor identityProcessor1,
                                    IdentityProcessor identityProcessor2) {
 
                     if (identityProcessor1.getPriority() > identityProcessor2.getPriority()) {
-                        return 1;
-                    } else if (identityProcessor1.getPriority() < identityProcessor2.getPriority()) {
                         return -1;
+                    } else if (identityProcessor1.getPriority() < identityProcessor2.getPriority()) {
+                        return 1;
```
Comment (johannnallathamby) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java]: According to API definition of Comparator this is what it says: "a negative integer, zero, or a positive integer as the first argument is less than, equal to, or greater than the second". Which means what is there in the code right now is correct isn't it?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/inbound/IdentityProcessor.java [hunk 4]
```diff
@@ -139,9 +140,12 @@ public void init(InitConfig initConfig) {
...
-        parameterMap.put(FrameworkConstants.SESSION_DATA_KEY, new String[] { sessionDataKey });
-        parameterMap.put(FrameworkConstants.RequestParams.TYPE, new String[] { getName() });
+        Map<String, String[]> modParaMap = new HashMap<String, String[]>();
+        for (String key : parameterMap.keySet()) {
+            modParaMap.put(key, parameterMap.get(key));
+        }
+        modParaMap.put(FrameworkConstants.SESSION_DATA_KEY, new String[]{sessionDataKey});
+        modParaMap.put(FrameworkConstants.RequestParams.TYPE, new String[]{getName()});
 
         authenticationRequest.appendRequestQueryParams(parameterMap);
```
Comment (johannnallathamby) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/inbound/IdentityProcessor.java]: Above changes are not required actually. The request parameters are passed to framework through AuthenticationRequest object in above line. It is put to cache and passed, so no need to pass as query parameters. Also passing as query parameters is vulnerable and that why we adopted the cache approach. The only issue that was there in my code is that I have unnecessarily tried to add FrameworkConstants.SESSION_DATA_KEY and FrameworkConstants.RequestParams.TYPE to an unmodifiable map which is not required at all.

**Comment by johannnallathamby** [comment]
Required improvements/changes have been merged with #220

**Pull request closed without merging**